lfs.c

```diff
@@ -2410,12 +2414,19 @@ int lfs_file_close(lfs_t *lfs, lfs_file_t *file) {
     // clean up memory
     if (!file->cfg->buffer) {
         lfs_free(file->cache.buffer);
+        file->cache.buffer = NULL;
```

Comment:
Is this needed with the file protection logic? Counter-intuitively it would actually hide use-after-free errors when tested inside valgrind.

Reply:
Indded protecttion logic is enough and this is not needed.
It is just result of my habit to assign NULL after deallocation.